test_rmw_implementation/test/test_publisher.cpp

```diff
@@ -375,3 +375,60 @@ TEST_F(CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION), count_mismatched_subscri
   EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
   EXPECT_EQ(0u, subscription_count);
 }
+
+class CLASSNAME (TestPublisherUseLoan, RMW_IMPLEMENTATION)
+  : public CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION)
+{
+protected:
+  using Base = CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION);
+
+  void SetUp() override
+  {
+    Base::SetUp();
+    // Test if Loaning is supported
+    void * msg_pointer{nullptr};
+    rmw_ret_t ret = rmw_borrow_loaned_message(pub, ts, &msg_pointer);
+    if (ret == RMW_RET_UNSUPPORTED) {
+      rmw_reset_error();
+      GTEST_SKIP();
+    } else {
+      FAIL() << "Tests not ready";
+    }
+    ret = rmw_return_loaned_message_from_publisher(pub, msg_pointer);
+    EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+  }
+
+  void TearDown() override
+  {
+    Base::TearDown();
+  }
+};
+
+TEST_F(CLASSNAME(TestPublisherUseLoan, RMW_IMPLEMENTATION), borrow_loaned_message_bad_arguments) {
+  void * msg_pointer = nullptr;
+  rmw_ret_t ret = rmw_borrow_loaned_message(nullptr, ts, &msg_pointer);
+  EXPECT_EQ(RMW_RET_INVALID_ARGUMENT, ret) << rmw_get_error_string().str;
```

Comment:
@Lobotuerk consider checking that `msg_pointer` remains `nullptr`. Same elsewhere.